### ISSUE TITLE
The number field compare filter does not always work correctly

### ISSUE THREAD

**Issue by vhar** [meilisearch/meilisearch#4374]
I have a numeric field in an array.
Sometimes after adding/updateing document on index when applying a filter I get an empty result.
Moreover, if you change the number in the filter, the result appears.
After update the document the second or third time the filter starts working correctly.
Add document:
```
curl \
  -X POST 'http://localhost:7700/indexes/tyres/documents' \
  -H 'Content-Type: application/json' \
  --data-binary '[
    {
      "ID":"f57d08fc-6089-11eb-8109-005056bc50fa",
      "Article":"1025873",
      "Width":235,
      "Height":55,
      "Diameter":18,
      "RegionCounts":{
        "42":40
      }
    }
  ]'
```
request
```
curl \ 
  -X POST 'http://localhost:7700/indexes/tyres/search' \
  -H 'Content-Type: application/json' \
  --data-binary '{
     "q": "",
     "filter": "RegionCounts.42 > 3"
 }'
```
return blank result

change filter to `RegionCounts.42 > 39` - have one hits result

change filter to `RegionCounts.42 > 31` have blank result

change filter to `RegionCounts.42 > 32` have one hits result

After update document all work fine, filter `RegionCounts.42 > 3` return one hits

**Comment by dureuill** [comment]
Hello @vhar :wave:

1. which version of Meilisearch are you using?
2. is the above sequence of commands sufficient to reproduce the issue? I think we already had similar reports but have been unable to reproduce so far, so a small self-contained reproducer would help us a lot.

thank you for the report

**Comment by irevoire** [comment]
Hey @vhar,

FYI I tried to reproduce your issue once again, and it didn't reproduce, see the following screenshot:
![image](https://github.com/meilisearch/meilisearch/assets/7032172/231606ef-18be-45f7-9342-c834c1fe7ab9)

**Comment by Kerollmops** [comment]
Hey @vhar 👋

Your issue is probably related to a bug. Unfortunately, it seems that similar issues were reported on https://github.com/meilisearch/meilisearch/discussions/4390 and https://github.com/meilisearch/meilisearch/issues/4339. We will investigate this next week.

However, to investigate, we need a complete reproduction example. What's the Meilisearch version you use? Do you have a script or something to send your settings and documents to Meilisearch? It's important to send them in batches and not give us only the documents, as it is probably related to updating the documents during the lifetime of the Meilisearch index.

Have a good day 🌻

**Comment by Kerollmops** [comment]
Hey @vhar 👋

Good news: We found the root cause of the issue and will land a v1.6.2 with a hotfix, which will be fixed in v1.7 cleaner. The bug is in the incremental part of the facets, and we removed it for the hotfix but will fix it correctly in v1.7.

Closing this in favor of #4409.

Thank you again for your report ❤️
